catalogue_pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedder.scala

```diff
@@ -46,15 +47,21 @@ class IdEmbedder @Inject()(metricsSender: MetricsSender,
   private def addIdentifierToJsonObject(obj: JsonObject): JsonObject = {
     if (obj.contains("sourceIdentifier")) {
       val sourceIdentifier = parseSourceIdentifier(obj)
-      val ontologyType = obj("ontologyType").get.asString.get
 
       val canonicalId = identifierGenerator
         .retrieveOrGenerateCanonicalId(
-          identifier = sourceIdentifier,
-          ontologyType = ontologyType
+          identifier = sourceIdentifier
         )
         .get
-      ("canonicalId", Json.fromString(canonicalId)) +: obj
+
+      if (obj.contains("type")) {
+        (("type", Json.fromString("Identified")) +:
```

Comment:
This makes me winge a bit but, with a model agnostic id_minter, I don't know how to avoid it

Reply:
It might help to override the discriminator defaults that Circe uses for sealed traits (class names) so that at least changing a class name won't break our JSON.

Reply:
I don't think you can...it uses shapeless under the hood and that's what shapeless gives you in a coproduct